src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -68,6 +68,8 @@ export type DeviceState = {
   recoveryEntropy: Entropy
   recoveryCharacterIndex: number | undefined
   recoveryWordIndex: number | undefined
+  isUpdatingPin: boolean | undefined
+  isLoading: boolean | undefined
```

Comment:
What does this represent? It feels closer to a property of the UI than the device.

Reply:
The KeepKey isn't telling the application that it's currently loading, so it should probably be used carefully but this is a state that I added in order to tell the UI that the keepkey is loading at this time

When you enter the new pin, for example, the keepkey enter a loading state, the UI needs to know it in order to inform the user that it's currently loading...